ktor-server/ktor-server-core/jvm/src/io/ktor/config/ApplicationConfig.kt

```diff
@@ -39,18 +44,160 @@ interface ApplicationConfig {
  * Represents an application config value
  */
 @KtorExperimentalAPI
-interface ApplicationConfigValue {
+class HoconApplicationConfigValue(private val config: Config, private val path: String) {
     /**
-     * Get property string value
+     * If the value is empty
+     * @see [Config.isEmpty]
      */
-    fun getString(): String
+    fun isEmpty(): Boolean = config.isEmpty
 
     /**
-     * Get property list value
+     * Get property [Boolean] value
+     * @see [Config.getBoolean]
      */
-    fun getList(): List<String>
+    fun getBoolean(): Boolean = config.getBoolean(path)
+
+    /**
+     * Get property [Number] value
+     * @see [Config.getNumber]
+     */
+    fun getNumber(): Number = config.getNumber(path)
+
+    /**
+     * Get property [Int] value
+     * @see [Config.getInt]
+     */
+    fun getInt(): Int = config.getInt(path)
```

Comment:
I believe these functions would be just extension functions like `getString().toInt()`. For sure there is no need to have members like that unless there is something very special in number representation in configs, right?

Reply:
HOCON actually loads the configurations with the correct types, so an integer is loaded as an integer and kept as such. Currently every time you get an `int` in a config, it gets converted to a `String`, and then you do `toInt()` to convert it back into an integer (or whatever other type).

You can test this by putting a breakpoint in `SimpleConfig::findKeyOrNull`. `v` will be of type `ConfigInt`, which then gets converted to a `Quoted` (which is a wrapper around a string) in `DefaultTransformer.transform`.